# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1517130345 -3600
#      Sun Jan 28 10:05:45 2018 +0100
# Node ID 41a87a2a8d381393ada16ebdef0907e57ba1ef24
# Parent  d36e1f768bfaa5fe4eb58c0152213cae3d8fc4d5
check current codec if it is valid before trying to use it (bug #52985)

* file-editor-tab.cc (check_valid_codec): check for valdi codec pointer
  before trying to encode the current text with it.

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -2287,16 +2287,27 @@ file_editor_tab::check_valid_identifier 
     }
 
   return false;
 }
 
 bool
 file_editor_tab::check_valid_codec (QTextCodec *codec)
 {
+  if (! codec)
+    {
+      int ans = QMessageBox::critical (nullptr,
+            tr ("Octave Editor"),
+            tr ("The current codec %1\n"
+                "can not be applied.\n\n"
+                "Please select another one or cancel saving!").arg (_encoding));
+
+      return true;
+    }
+
   if (! codec->canEncode (_edit_area->text ()))
     {
       int ans = QMessageBox::warning (nullptr,
             tr ("Octave Editor"),
             tr ("The current editor contents can not be encoded\n"
                 "with the selected codec %1.\n"
                 "Using it will result in data loss!\n\n"
                 "Do you want to chose another codec?").arg (_encoding),
